### PR TITLE
Feature/get options for dimension

### PR DESCRIPTION
**Pull request by mohsdev**
### What

Needs to get a list of dimension options available for a given Cantabular Blob.
Dimension options will consist of a name, code_list, code and version.
The response from this endpoint will list the dimension options for a cantabular blob.

### How to review

Make sure code changes make sense
Make sure unit tests pass

### Who can review

Anyone

### CODE REVIEW

File: handler/dimension_options.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	var request contract.GetDimensionOptionsRequest
+	if err := schema.NewDecoder().Decode(&request, r.URL.Query()); err != nil {
+		s.respond.Error(
+			ctx,
+			w,
+			http.StatusBadRequest,
+			errors.Wrap(err, "failed to decode query parameters"),
+		)
+		return
+	}
```
Comment (PauloPortugal) [inline handler/dimension_options.go]: We are not using query parameters on this endpoint. Please remove this.

File: contract/dimension_options.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package contract
+
+import "github.com/ONSdigital/dp-cantabular-dimension-api/model"
+
+// GetDimensionOptionsRequest defines the schema for the GET /dimension/{dimension}/options query parameter
+type GetDimensionOptionsRequest struct {
+	Dimension string `json:"dimension"`
+}
+
```
Comment (PauloPortugal) [inline contract/dimension_options.go]: I don't this is required or needed.

File: contract/dimension_options.go [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+// GetDimensionOptionsResponse is the response object for GET /dimension/{dimension}/options
+type GetDimensionOptionsResponse struct {
+	Options []*model.DimensionOption `json:"options"`
```
Comment (PauloPortugal) [inline contract/dimension_options.go]: Can `options` be `nil`?
If instead we should return an empty list/slice/array then remove the `pointer`

File: handler/dimension_options.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+	cantabularRequest := cantabular.GetDimensionOptionsRequest{
+		DimensionNames: []string{request.Dimension},
+	}
```
Comment (PauloPortugal) [inline handler/dimension_options.go]: Isn't `request.Dimension` empty here?
Please review this.

File: handler/dimension_options.go [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+	cantabularResponse, err := s.ctblr.GetDimensionOptions(ctx, cantabularRequest)
```
Comment (PauloPortugal) [inline handler/dimension_options.go:36]: Is this correct?

File: model/dimention_option.go [hunk 6]
```diff
@@ -0,0 +1,20 @@
+package model
+
+// LinkObject represents a generic structure for all links
+type LinkObject struct {
```
Comment (PauloPortugal) [inline model/dimention_option.go]: I would call it Link rather as Golang does not have `objects` as found on C#/C++/Java/Scala.

**Review by PauloPortugal — CHANGES_REQUESTED** [review]
Please review the comments.

Also we should also update the PR description instead of just relying on the template version.

Also you will need to write component tests to provide the minimum coverage to this new endpoint, which should include Happy path and unhappy paths please.

The implementation does not seem entirely aligned with the ticket.

File: model/dimention_option.go (hunk 6 above)
Comment (mohsdev) [reply model/dimention_option.go]: done

File: contract/dimension_options.go (hunk 3 above)
Comment (mohsdev) [reply contract/dimension_options.go]: done

File: handler/dimension_options.go (hunk 1 above)
Comment (mohsdev) [reply handler/dimension_options.go]: this is an api for "GET /dimension/{dimension}/options", i think the query parameter is needed?

Comment (mohsdev) [reply handler/dimension_options.go]: done

File: contract/dimension_options.go (hunk 2 above)
Comment (mohsdev) [reply contract/dimension_options.go]: done

File: handler/dimension_options.go (hunk 4 above)
Comment (mohsdev) [reply handler/dimension_options.go]: done

**Comment by andre-urbani** [comment]
Repository is being decommissioned
